src/assisted_test_infra/test_infra/helper_classes/cluster.py

```diff
@@ -801,10 +803,46 @@ def download_image(self, iso_download_path: str = None) -> Path:
             )
         return self._infra_env.download_image(iso_download_path)
 
+    # Downloads the iPXE script from assisted service to a local file
+    def prepare_local_ipxe_server(self, infra_env_id):
+        assisted_service_url = utils.get_assisted_service_url_by_configmap("assisted-service-config", "assisted-installer")
+        remote_ipxe_path = f"/api/assisted-install/v2/infra-envs/{infra_env_id}/downloads/files?file_name=ipxe-script"
```

Comment:
we decided a while ago doing all communication with the service via ``assisted_service_api.py`` script, and using the automatically generated one
That's mainly to have a single point for abstracting-away this kind of communication, and I think this method (or maybe part of it) should reside in the "adapter"

Reply:
Moved some of the logic to the `assisted_service_api.py` file under the `download_and_save_infra_env_file` function. How does it look?